docs/source/customizations/overview.mdx

```diff
@@ -276,6 +276,23 @@ flowchart LR;
 
 Additionally, some requests and responses may happen multiple times for the same operation. With subscriptions, for example, a subgraph sends a new `SubgraphResponse` whenever data is updated. Each response object travels through all the services in the response path and interacts with any customizations you've created.
 
+### Request and Response Buffering
+
+The router expects to execute on a stream of data. In order to work correctly and provide high performance, the following expectations must be met:
+
+##### Request Path
+__No buffering before the end of the `router_service` processing step__
+
+##### Response Path
+__No buffering__
+
+> In general, it's best to avoid buffering where possible. It is "ok" to do so on the request path once the `router_service` step is complete.
```

Comment:
Can we give a reason for "avoid buffering possible" here? Just "increased latency" or "decreased performance" or something similar, so the reader understands what their trade-off is and when it's actually "ok" to do so.

```suggestion
> In general, it's best to avoid buffering where possible. If necessary, it is ok to do so on the request path once the `router_service` step is complete.
```